### PR TITLE
#106 new LSP settings for 2016/2019

### LINKED ISSUE #106: Add LSP settings introduced in 2016/2019
The following settings were newly introduced with Windows 2016 or 2019 and need to be added to lsp_mapping:

- DCOM: Machine Access Restrictions in Security Descriptor Definition Language (SDDL) syntax
- DCOM: Machine Launch Restrictions in Security Descriptor Definition Language (SDDL) syntax
- Network security: Allow LocalSystem NULL session fallback
- Network security: Allow PKU2U authentication requests to this computer to use online identities.
- Network security: Restrict NTLM: Add remote server exceptions for NTLM authentication
- Network security: Restrict NTLM: Add server exceptions in this domain
- Network security: Restrict NTLM: Audit Incoming NTLM Traffic
- Network security: Restrict NTLM: Audit NTLM authentication in this domain
- Network security: Restrict NTLM: Incoming NTLM traffic
- Network security: Restrict NTLM: NTLM authentication in this domain
- Network security: Restrict NTLM: Outgoing NTLM traffic to remote servers
- Network access: Restrict clients allowed to make remote calls to SAM
- Interactive logon: Don't display username at sign-in
- Interactive logon: Require Windows Hello for Business or smart card
- Microsoft network server: Attempt S4U2Self to obtain claim information

### PR DESCRIPTION
**Pull request by AndyAdrian-OP**
Adds LSP entries introducted in Windows 2016 and 2019.
Updates README with same.

closes #106

### CODE REVIEW

File: lib/puppet_x/lsp/security_policy.rb [hunk 1]
```diff
@@ -849,7 +849,86 @@ def self.lsp_mapping
...
+        name: 'MACHINE\System\CurrentControlSet\Services\Netlogon\Parameters\RestrictNTLMInDomain',
+        policy_type: 'Registry Values',
+        reg_type: '4',
+      },
+      'Network security: Restrict NTLM: Outgoing NTLM traffic to remote servers' => {
+        name: 'MACHINE\System\CurrentControlSet\Control\Lsa\MSV1_0\RestrictSendingNTLMTraffic',
+        policy_type: 'Registry Values',
+        reg_type: '4',
+      },
+      'Interactive logon: Don\'t display last signed-in' => {
```
Comment (ayohrling) [inline lib/puppet_x/lsp/security_policy.rb:917]: Is this accurate escaping with single-quotes? Would it be more accurate/readable to utilize no escaping but double-quote the whole key here?

```
"Interactive logon: Don't display last signed-in" => {
```

File: lib/puppet_x/lsp/security_policy.rb [hunk 2]
```diff
@@ -849,7 +849,86 @@ def self.lsp_mapping
...
+        name: 'MACHINE\Software\Microsoft\Windows\CurrentVersion\Policies\System\DontDisplayLastUserName',
+        reg_type: '4',
+        policy_type: 'Registry Values',
+      },
+      'Interactive logon: Don\'t display username at sign-in' => {
```
Comment (ayohrling) [inline lib/puppet_x/lsp/security_policy.rb:922]: Same question posed as line 917

**Review by ayohrling — COMMENTED** [review]
Just a couple questions regarding escaping and single quotes within the LSP mapped hash.

File: lib/puppet_x/lsp/security_policy.rb (hunk 1 above)
Comment (AndyAdrian-OP) [reply lib/puppet_x/lsp/security_policy.rb:917]: I expected Rubocop to complain about double-quoting a static string, but it doesn't in this case apparently.  I'll push changes.

File: lib/puppet_x/lsp/security_policy.rb (hunk 2 above)
Comment (AndyAdrian-OP) [reply lib/puppet_x/lsp/security_policy.rb:922]: same as above.

**Review by ayohrling — APPROVED** [review]

**Pull request merged**